examples/guestbook/README.md

```diff
@@ -103,20 +128,96 @@ spec:
 [Download example](redis-master-controller.yaml?raw=true)
 <!-- END MUNGE: EXAMPLE redis-master-controller.yaml -->
 
-Change to the `<kubernetes>/examples/guestbook` directory if you're not already there. Create the redis master pod in your Kubernetes cluster by running:
+A Kubernetes [service](../../docs/user-guide/services.md) is a named load balancer that proxies traffic to one or more containers. This is done using the [labels](../../docs/user-guide/labels.md) metadata that we defined in the `redis-master` pod above.  As mentioned, we have only one redis master, but we nevertheless want to create a service for it. Why? Because it gives us a deterministic way to route to the single master using an elastic IP.
+
+Services find the pods to load balance based on the pods' labels.
+The selector field of the service description determines which pods will receive the traffic sent to the service, and the `port` and `targetPort` information defines what port the service proxy will run at.
+
+The file `examples/guestbook/redis-master-service.yaml` defines the redis master service:
+
+<!-- BEGIN MUNGE: EXAMPLE redis-master-service.yaml -->
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-master
+  labels:
+    app: redis
+    tier: backend
+    role: master
+spec:
+  ports:
+    # the port that this service should serve on
+  - port: 6379
+    targetPort: 6379
+  selector:
+    app: redis
+    tier: backend
+    role: master
+```
+
+[Download example](redis-master-service.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE redis-master-service.yaml -->
+
+First create the service by running:
```

Comment:
It'd be nice to inform the user why it's better to create a service first. 
(See https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/config-best-practices.md)